### ISSUE TITLE
Add SQL parser support for PostgreSQL ILIKE

### ISSUE THREAD

**Issue by MirkoTeran** [jOOQ/jOOQ#8451]
### Use case:

I would like to translate SQL queries written for Postgres to other databases. Currently it doesn't support ILIKE (https://www.postgresql.org/docs/11/functions-matching.html) operation.

Example:

```sql
select 
  trim(p_name) as "Name", 
  p_street as "Street", 
  p_city as "City", 
  p_zipcode as "Zip Code"
from dimpartner 
where p_name ilike '%name%'
and p_zipcode ilike '%4000%'
order by trim(p_name) asc
```

results in 

```
Unexpected token or missing query delimiter: [7:14] ...pcode as "Zip Code"
from dimpartner 
where p_name [*]ilike '%name%'
and p_zipcode ilike '%4000%'
order by trim(p_name) asc
```

### Possible solution you'd like to see:

Support for ILIKE syntax.

### Possible workarounds:

Using combination of like and upper/lower.

### Versions:

- jOOQ: 3.11.10 pro (also tested with https://www.jooq.org/translate/ - same results)
- Java: 11
- Database (include vendor): Postgres 9.6 - 11
- OS: Win10
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks a lot for your request. Indeed, we already generate `ilike` predicates when using `Field.likeIgnoreCase()`, which is a more Java-ish way of saying `ilike`. The current emulation is comparing `lower(left) like lower(right)`, which is not 100% accurate in more esoteric collations, but might just be good enough.

For consistency, I suspect we can add this functionality in the parser.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. We'll update https://www.jooq.org/translate soon to reflect this change.

**Comment by MirkoTeran** [comment]
Thank you for super fast update!
